In [ ]:
%pip install python-dotenv langchain-upstage

  Using cached langchain_upstage-0.7.7-py3-none-any.whl.metadata (3.3 kB)
  Using cached pypdf-6.10.2-py3-none-any.whl.metadata (7.1 kB)
  Using cached tokenizers-0.20.3.tar.gz (340 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
  Using cached huggingface_hub-0.36.2-py3-none-any.whl.metadata (15 kB)
  Using cached filelock-3.29.0-py3-none-any.whl.metadata (2.0 kB)
  Using cached fsspec-2026.3.0-py3-none-any.whl.metadata (10 kB)
  Using cached hf_xet-1.4.3-cp37-abi3-macosx_11_0_arm64.whl.metadata (4.9 kB)
Using cached langchain_upstage-0.7.7-py3-none-any.whl (26 kB)
Using cached pypdf-6.10.2-py3-none-any.whl (336 kB)
Using cached huggingface_hub-0.36.2-py3-none-any.whl (566 kB)
Using cached hf_xet-1.4.3-cp37-abi3-macosx_11_0_arm64.whl (3.6 MB)
Using cached fsspec-2026.3.0-py3-none-any.whl (202 kB)
Using cached filelock-3.29.0-py3-none-any.whl (39 kB)
  e

In [ ]:
%pip uninstall python-dotenv langchain-upstage

Found existing installation: python-dotenv 1.2.2
Uninstalling python-dotenv-1.2.2:
  Would remove:
    /Users/leehk/.pyenv/versions/3.14.4/envs/tutorial-langchain/bin/dotenv
    /Users/leehk/.pyenv/versions/3.14.4/envs/tutorial-langchain/lib/python3.14/site-packages/dotenv/*
    /Users/leehk/.pyenv/versions/3.14.4/envs/tutorial-langchain/lib/python3.14/site-packages/python_dotenv-1.2.2.dist-info/*
Proceed (Y/n)? ^C
Note: you may need to restart the kernel to use updated packages.


In [ ]:
%pip uninstall python-dotenv langchain-upstage -y

Found existing installation: python-dotenv 1.2.2
Uninstalling python-dotenv-1.2.2:
  Successfully uninstalled python-dotenv-1.2.2
Note: you may need to restart the kernel to use updated packages.


In [3]:
%uv pip install langchain-upstage

/Users/leehk/.pyenv/versions/tutorial-langchain/bin/python: No module named uv
Note: you may need to restart the kernel to use updated packages.


In [1]:
from dotenv import load_dotenv

load_dotenv()

ModuleNotFoundError: No module named 'dotenv'

In [2]:
%pip install python-dotenv

  Using cached python_dotenv-1.2.2-py3-none-any.whl.metadata (27 kB)
Using cached python_dotenv-1.2.2-py3-none-any.whl (22 kB)
Note: you may need to restart the kernel to use updated packages.


In [4]:
from langchain_upstage import ChatUpstage

llm = ChatUpstage()

ModuleNotFoundError: No module named 'langchain_upstage'

In [5]:
%pip install uv

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 21.8/21.8 MB 4.8 MB/s  0:00:04m0:00:0100:01
Note: you may need to restart the kernel to use updated packages.


In [7]:
%uv pip install langchain-upstage

Using Python 3.14.4 environment at: /Users/leehk/.pyenv/versions/tutorial-langchain
Resolved 41 packages in 872ms                                        
⠙ Preparing packages... (0/7)                                                   
⠙ Preparing packages... (0/7)-------------------     0 B/197.85 KiB          
⠙ Preparing packages... (0/7)-------------------     0 B/197.85 KiB          
fsspec               ------------------------------     0 B/197.85 KiB
⠙ Preparing packages... (0/7)-------------------     0 B/328.43 KiB          
fsspec               ------------------------------     0 B/197.85 KiB
⠙ Preparing packages... (0/7)-------------------     0 B/328.43 KiB          
fsspec               ------------------------------     0 B/197.85 KiB
⠙ Preparing packages... (0/7)-------------------     0 B/328.43 KiB          
filelock             ------------------------------     0 B/38.88 KiB
fsspec               ------------------------------     0 B/197.85 KiB
⠙ Preparing packages.

In [13]:
from dotenv import load_dotenv

load_dotenv()


True

In [14]:
from langchain_upstage import ChatUpstage

ModuleNotFoundError: No module named 'langchain_upstage'

In [12]:
%uv pip install python-dotenv

Using Python 3.14.4 environment at: /Users/leehk/.pyenv/versions/tutorial-langchain
Checked 1 package in 3ms
Note: you may need to restart the kernel to use updated packages.


In [15]:
%pip install -qU langchain-core langchain-upstage

  error: subprocess-exited-with-error
  
  × Building wheel for tokenizers (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [158 lines of output]
      Rust not found, installing into a temporary directory
      Python reports SOABI: cpython-314-darwin
      Computed rustc target triple: aarch64-apple-darwin
      Installation directory: /Users/leehk/Library/Caches/puccinialin
      Rustup already downloaded
      Installing rust to /Users/leehk/Library/Caches/puccinialin/rustup
      warn: It looks like you have an existing rustup settings file at:
      warn: /Users/leehk/Library/Caches/puccinialin/rustup/settings.toml
      warn: Rustup will install the default toolchain as specified in the settings file,
      warn: instead of the one inferred from the default host triple.
      info: profile set to minimal
      info: setting default host triple to aarch64-apple-darwin
      warn: Updating existing toolchain, profile choice will be ignored
      info: syncing chan

In [16]:
%uv pip install -qU langchain-core langchain-upstage

  × Failed to download and build `tokenizers==0.20.3`
  ├─▶ Failed to parse:
  │   `/Users/leehk/.cache/uv/sdists-v9/pypi/tokenizers/0.20.3/RQAhbAaYb6fHg9j76YGeG/src/pyproject.toml`
  ╰─▶ TOML parse error at line 1, column 1
        |
      1 | [project]
        | ^^^^^^^^^
      `pyproject.toml` is using the `[project]` table, but the required
      `project.version` field is neither set nor present in the
      `project.dynamic` list

  help: `tokenizers` (v0.20.3) was included because `langchain-upstage`
        (v0.7.7) depends on `tokenizers`
Note: you may need to restart the kernel to use updated packages.


In [20]:
%uv pip install "tokenizers>0.20.3"

Using Python 3.14.4 environment at: /Users/leehk/.pyenv/versions/tutorial-langchain
Resolved 23 packages in 1.25s                                        
⠙ Preparing packages... (0/8)                                                   
⠙ Preparing packages... (0/8)-------------------     0 B/9.53 KiB            
⠙ Preparing packages... (0/8)-------------------     0 B/9.53 KiB            
annotated-doc        ------------------------------     0 B/5.18 KiB
⠙ Preparing packages... (0/8)-------------------     0 B/9.53 KiB            
annotated-doc        ------------------------------     0 B/5.18 KiB
⠙ Preparing packages... (0/8)---------- 9.53 KiB/9.53 KiB           
annotated-doc        ------------------------------     0 B/5.18 KiB
⠙ Preparing packages... (0/8)---------- 9.53 KiB/9.53 KiB           
⠙ Preparing packages... (0/8)-------------------     0 B/5.18 KiB            
⠙ Preparing packages... (0/8)---------- 5.18 KiB/5.18 KiB           
⠙ Preparing packages... (0/8)----------

In [21]:
%uv pip install langchain-core langchain-upstage

Using Python 3.14.4 environment at: /Users/leehk/.pyenv/versions/tutorial-langchain
Resolved 41 packages in 345ms                                        
   Building tokenizers==0.20.3                                         
  × Failed to download and build `tokenizers==0.20.3`                                                
  ├─▶ Failed to parse:
  │   `/Users/leehk/.cache/uv/sdists-v9/pypi/tokenizers/0.20.3/RQAhbAaYb6fHg9j76YGeG/src/pyproject.toml`
  ╰─▶ TOML parse error at line 1, column 1
        |
      1 | [project]
        | ^^^^^^^^^
      `pyproject.toml` is using the `[project]` table, but the required
      `project.version` field is neither set nor present in the
      `project.dynamic` list

  help: `tokenizers` (v0.20.3) was included because `langchain-upstage`
        (v0.7.7) depends on `tokenizers`
Note: you may need to restart the kernel to use updated packages.


In [22]:
%pip install -U langchain-core langchain-upstage

  Using cached langchain_core-1.3.0-py3-none-any.whl.metadata (4.4 kB)
  Using cached langchain_upstage-0.7.7-py3-none-any.whl.metadata (3.3 kB)
  Using cached pypdf-6.10.2-py3-none-any.whl.metadata (7.1 kB)
  Using cached tokenizers-0.20.3.tar.gz (340 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
  Using cached huggingface_hub-0.36.2-py3-none-any.whl.metadata (15 kB)
Using cached langchain_core-1.3.0-py3-none-any.whl (515 kB)
Using cached langchain_upstage-0.7.7-py3-none-any.whl (26 kB)
Using cached pypdf-6.10.2-py3-none-any.whl (336 kB)
Using cached huggingface_hub-0.36.2-py3-none-any.whl (566 kB)
  error: subprocess-exited-with-error
  
  × Building wheel for tokenizers (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [158 lines of output]
      Rust not found, installing into a temporary directory
      Python reports SOABI: cpytho

# Upstage install
- pip, uv 문제가 아니라 파이썬 문제로 파이썬 버전을 다운그레이드 해야 함

> error: the configured Python interpreter version (3.14) is newer than PyO3's maximum supported version (3.13)

In [27]:
%pip uninstall python-dotenv langchain-core langchain-openai -y

Found existing installation: python-dotenv 1.2.2
Uninstalling python-dotenv-1.2.2:
  Successfully uninstalled python-dotenv-1.2.2
Found existing installation: langchain-openai 1.1.16
Uninstalling langchain-openai-1.1.16:
  Successfully uninstalled langchain-openai-1.1.16
Note: you may need to restart the kernel to use updated packages.


In [2]:
%pip install python-dotenv langchain-upstage

  Using cached langchain_upstage-0.7.7-py3-none-any.whl.metadata (3.3 kB)
  Using cached pypdf-6.10.2-py3-none-any.whl.metadata (7.1 kB)
  Using cached huggingface_hub-0.36.2-py3-none-any.whl.metadata (15 kB)
  Using cached filelock-3.29.0-py3-none-any.whl.metadata (2.0 kB)
  Using cached fsspec-2026.3.0-py3-none-any.whl.metadata (10 kB)
  Using cached hf_xet-1.4.3-cp37-abi3-macosx_11_0_arm64.whl.metadata (4.9 kB)
Using cached langchain_upstage-0.7.7-py3-none-any.whl (26 kB)
Using cached pypdf-6.10.2-py3-none-any.whl (336 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.6/2.6 MB 23.4 MB/s  0:00:00
Using cached huggingface_hub-0.36.2-py3-none-any.whl (566 kB)
Using cached hf_xet-1.4.3-cp37-abi3-macosx_11_0_arm64.whl (3.6 MB)
Using cached fsspec-2026.3.0-py3-none-any.whl (202 kB)
Using cached filelock-3.29.0-py3-none-any.whl (39 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 7/7 [langchain-upstage]huggingface-hub]
Note: you may need to restart the kernel to use updated packages.


In [1]:
from dotenv import load_dotenv

load_dotenv()

True

In [2]:
from langchain_upstage import ChatUpstage

llm = ChatUpstage()

In [3]:
response = llm.invoke("하이닉스 오늘 실적 발표 요약해줘.")

In [4]:
response.content

'하이닉스는 2022년 4분기 실적을 오늘 발표했습니다. 주요 내용은 다음과 같습니다.\n\n1. **매출**: 13조 77억원으로, 이전 분기 대비 7% 감소했지만, 전년 동기 대비 36% 증가했습니다.\n\n2. **영업이익**: 1조 7천억 원으로, 이전 분기 대비 33% 감소했습니다.\n\n3. **순이익**: 9천억 원으로, 이전 분기 대비 43% 감소했습니다.\n\n4. **제품별 실적**:\n   - **DRAM**: 평균 판매가격이 21% 하락했으며, 출하량은 26% 증가했습니다.\n   - **NAND**: 평균 판매가격이 17% 하락했고, 출하량은 10% 증가했습니다.\n\n5. **사업 전망**:\n   - AI 및 고성능 컴퓨팅(HPC) 분야에서의 수요 증가를 기대하며, 이러한 분야의 제품 개발과 공급 확대에 주력할 계획입니다.\n   - 또한, 온디바이스 AI와 같은 새로운 분야에서도 기회를 모색할 예정입니다.\n\n6. **투자 계획**:\n   - 올해 시설투자에 5조 원대를 집행할 계획입니다.\n   - 이중 메모리반도체에는 4조 원, 시스템IC에는 1조 원을 투자할 예정입니다.\n\n하이닉스는 어려운 시장 환경에도 불구하고 기술 및 원가 경쟁력에서 우위를 유지하고, AI 및 HPC 분야에서의 성장을 기대하고 있습니다.'